lib/rules/sort-destructuring-keys.js

```diff
@@ -82,9 +143,15 @@ module.exports = {
     return {
       ObjectPattern( node ) {
         const { properties } = node
-        const keys = properties
+        const filteredProperties = properties
           .filter(prop => !prop.computed && prop.type !== 'ExperimentalRestProperty')
-          .map(prop => getKeyName(prop))
+        const allKeys = filteredProperties.map(getKeyName)
+        const keys = filteredProperties
+          .map(( prop, i ) => [
+            allKeys[i],
+            getKeyValue(prop)
+              .filter(Array.prototype.includes, allKeys),
```

Comment:
Nice!

Reply:
Dang, this won't work properly in some cases, since `Array.prototype.includes` takes index as a second argument... and in this particular case it will receive index, since `filter` does have an index as its second argument...
Will need to fix that as well.